test/Constraints/protocols.swift

```diff
@@ -184,17 +184,17 @@ func staticExistential(_ p: P.Type, pp: P.Protocol) {
 
   let ppp: P = p.init()
 
-  _ = pp() // expected-error{{value of type 'P.Protocol' is a protocol; it cannot be instantiated}}
-  _ = pp().bar // expected-error{{value of type 'P.Protocol' is a protocol; it cannot be instantiated}}
-  _ = pp().bar(2) // expected-error{{value of type 'P.Protocol' is a protocol; it cannot be instantiated}}
+  _ = pp() // expected-error{{instance member 'init' cannot be used on type 'P'}}
+  _ = pp().bar // expected-error{{instance member 'init' cannot be used on type 'P'}}
+  _ = pp().bar(2) // expected-error{{instance member 'init' cannot be used on type 'P'}}
 
-  _ = pp.init() // expected-error{{protocol type 'P' cannot be instantiated}}
-  _ = pp.init().bar // expected-error{{protocol type 'P' cannot be instantiated}}
-  _ = pp.init().bar(3) // expected-error{{protocol type 'P' cannot be instantiated}}
+  _ = pp.init() // expected-error{{instance member 'init' cannot be used on type 'P'}}
+  _ = pp.init().bar // expected-error{{instance member 'init' cannot be used on type 'P'}}
+  _ = pp.init().bar(3) // expected-error{{instance member 'init' cannot be used on type 'P'}}
 
-  _ = P() // expected-error{{protocol type 'P' cannot be instantiated}}
-  _ = P().bar // expected-error{{protocol type 'P' cannot be instantiated}}
-  _ = P().bar(4) // expected-error{{protocol type 'P' cannot be instantiated}}
+  _ = P() // expected-error{{instance member 'init' cannot be used on type 'P'}}
+  _ = P().bar // expected-error{{instance member 'init' cannot be used on type 'P'}}
+  _ = P().bar(4) // expected-error{{instance member 'init' cannot be used on type 'P'}}
```

Comment:
All these problems would benefit greatly from https://github.com/apple/swift/pull/22379. It would be great if we could expect the use of `AllowInvalidInitRef::onProtocolMetatype` introduced there to fix this case.

Reply:
How do we do that?

Reply:
I think it could be done in `simplifyMemberConstraint`. If this is a call on metatype and `baseTy->is<MetatypeType>() && baseTy->getMetatypeInstanceType()->isAnyExistentialType()` and the unviable declarations are `ConstructorDecl *`, it's equivalent to writing:

```swift
protocol P {}
_ = P() <- 
```

In that case you can record `AllowInvalidInitRef::onProtocolMetatype` fix and it should do the right thing.

Reply:
@xedin Thanks for your suggestion! `onProtocolMetatype` fixes this case 🎉 Can you take a look?